lib/stripe_mock/request_handlers/events.rb

```diff
@@ -13,9 +13,36 @@ def retrieve_event(route, method_url, params, headers)
       end
 
       def list_events(route, method_url, params, headers)
-        Data.mock_list_object(events.values, params)
+        values = filter_by_created(events.values, params: params)
+        Data.mock_list_object(values, params)
       end
-      
+
+      private
+
+      def filter_by_created(event_list, params:)
+        if params[:created].nil?
+          return event_list
+        end
+
+        if params[:created].is_a?(Hash)
+          if params[:created][:gt]
+            event_list = event_list.select { |event| event[:created] > params[:created][:gt].to_i }
+          end
+          if params[:created][:gte]
+            event_list = event_list.select { |event| event[:created] >= params[:created][:gte].to_i }
+          end
+          if params[:created][:lt]
+            event_list = event_list.select { |event| event[:created] < params[:created][:lt].to_i }
+          end
+          if params[:created][:lte]
+            event_list = event_list.select { |event| event[:created] <= params[:created][:lte].to_i }
+          end
+        else
+          event_list = event_list.select { |event| event[:created] == params[:created].to_i }
+        end
+        event_list
```

Comment:
nit: you could remove the assignments and the last `event_list` so the if statement returns the value, like I had in my original suggestion. It's a bit less repetitive, but if you think this is more readable I don't care strongly.

Reply:
These `if`s don't return, they continue changing the result. You can apply multiple filters and all are considered when filtering results, e.g. `created: {gt: time1, lt: time2}` will only give you events between those two timestamps.

Reply:
Ah you're right, I misread the code.